internal/verify.go

```diff
@@ -0,0 +1,75 @@
+package internal
+
+import (
+	"encoding/base64"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/rs/zerolog"
+	"golang.org/x/crypto/nacl/sign"
+)
+
+// verify the signed data (value) of the set request body
+func verifySignedData(logger zerolog.Logger, data string, pk []byte) (bool, error) {
```

Comment:
we shouldn't pass logger to such `pure` functions, let's remove all the debugging statements in this function